codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2Connection.java

```diff
@@ -1165,4 +1132,36 @@ private boolean allowModifications() {
             return pendingIterations == 0;
         }
     }
+
+    /**
+     * Implementation of {@link PropertyKey} that specifies the index position of the property.
+     */
+    class DefaultPropertyKey implements PropertyKey {
+        private final int index;
+
+        DefaultPropertyKey(int index) {
+            this.index = index;
+        }
+    }
+
+    /**
+     * A registry of all stream property keys known by this connection.
+     */
+    private class PropertyKeyRegistry {
+        final List<DefaultPropertyKey> keys = new ArrayList<DefaultPropertyKey>(4);
```

Comment:
Just thought I would pitch it again here....one way to support "static" use cases would be:

1. Use an `IntObjectMap` here.
2. Define some static variable (backed by an enum or something) which provides the equivalent of `private final int END_STATIC_PROPERTY_KEY=x;
3. Define a `private final int nextKey = LAST_STATIC_PROPERTY_KEY;`
4. Instead of `DefaultPropertyKey key = new DefaultPropertyKey(keys.size());` we do `DefaultPropertyKey key = new DefaultPropertyKey(nextKey++);`